securedrop_client/gui/login_dialog.py

```diff
@@ -107,10 +107,10 @@ def __init__(self, parent: QWidget) -> None:
         buttons_layout = QHBoxLayout()
         buttons.setLayout(buttons_layout)
         buttons_layout.setContentsMargins(0, 20, 0, 0)
-        self.submit = SignInButton()
+        self.submit = SignInButton(self)
         self.submit.setDefault(True)
```

Comment:
This is where the "submit" button is made the default, but that only has meaning if the button's `parent` is actually a `QDialog` (or one of them, parenting in Qt is a chain). In our case, `self` is indeed a `QDialog` but the relationship wasn't established.

Note that in general, it is good to specify the widgets' parents because that informs the garbage-collector's work.

Reply:
parent_layout.addWidget(child)  # this does `child.setParent` for you, automatically

Reply:
you could do this, but you'd have to make sure to change the default button from the offline mode button to the submit button here:

```
diff --git a/securedrop_client/gui/login_dialog.py b/securedrop_client/gui/login_dialog.py
index 23b7b11..4718b22 100644
--- a/securedrop_client/gui/login_dialog.py
+++ b/securedrop_client/gui/login_dialog.py
@@ -108,7 +108,6 @@ class LoginDialog(QDialog):
         buttons.setLayout(buttons_layout)
         buttons_layout.setContentsMargins(0, 20, 0, 0)
         self.submit = SignInButton()
-        self.submit.setDefault(True)
         self.submit.clicked.connect(self.validate)
         self.offline_mode = LoginOfflineLink()
         buttons_layout.addWidget(self.offline_mode)
@@ -137,6 +136,8 @@ class LoginDialog(QDialog):
         layout.addStretch()
         layout.addWidget(application_version)
 
+        self.submit.setDefault(True)
+
     def setup(self, controller: Controller) -> None:
         self.controller = controller
         self.offline_mode.clicked.connect(self.controller.login_offline_mode)
```
Personally, I think the code is cleaner when we rely on the framework to auto-assign the parent, so I'd like to better understand why setting the submit button to the default isn't honored when it's set as default before the button is parented.

Reply:
I see, since the default button is determined based on widget position in the layout, we have to wait until _after_ the layout is set before we override the default button. I think either implementation is fine, so not treating this as a blocker.

Reply:
> Personally, I think the code is cleaner when we rely on the framework to auto-assign the parent, so I'd like to better understand why setting the submit button to the default isn't honored when it's set as default before the button is parented.

Paraphrasing the docs: buttons that are children of a `QDialog` are automatically set in "autodefault" mode. The one that becomes the effective default is the first one *in the focus chain* at any given time.

I think of the default action as a "semantic" feature: we make a conscious choice as UX designers to pick what the default action should be. For example: it could be "Cancel" in the case of destructive operations, and "Ok" when the operation is easily reversible. 
The position of the buttons, on the other hand, is often independent of those considerations. Most operating systems will indeed assign specific positions to their "Ok" and "Cancel" buttons for consistency / familiarity purposes.

That's why I think that assigning the default button explicitly is not only fine, but preferable to relying on autofocus alone. Does thismake sense you you @creviera? How do you think about this?

Reply:
I don't think we're disagreeing here. As you see in the diff above, you could have accomplished this fix by just moving the `self.submit.setDefault(True)` after you add the widget to `layout`. Either way is fine, so I was just pointing out how you can rely on the framework to do the parenting for you. See you the diff fixes the issue? 

I think the default button here makes the most sense to be the "Submit" button, so no matter what we will need to assign that to be the default.

Reply:
Ooooh, apologies @creviera !! I read "assign the parent" and understood "assign the default" 🤦‍♀️

You're totally right, and the resulting code is simpler, I'll follow up with your suggestion.

For future reference:

> This is where the "submit" button is made the default, but that only has meaning if the button's parent is actually a QDialog (or one of them, parenting in Qt is a chain). In our case, self is indeed a QDialog **but the relationship wasn't established**.

It's not that the relationship was not established, the `QDialog` was in the buttons' parent **chain** from the moment they are added to the layout (which itself has the `QDialog` in its oarent chain), but there is an sequencing subtlety: setting the default takes effect _if_ the parenting relationship with `QDialog` is established _when `setDefault()` is called_. I didn't expect that, but could make sense because ultimately the key event is received _by the dialog_, not the button, so something is probably done to the dialog if oresent in the parent chain when the method is called in the button.

Thank you so much @creviera that's a great insight!! (And sorry for the misunderstanding, my bad entirely!)

Reply:
Addressed in #1342 - actually I'd say fixed, because it was worse than I thought! (The way I did t in this PR, the buttons' parent was changing from being `self` to being the layout, and even though that didn't break anything I think it was unnecessarily messy!)